### PR TITLE
Add initial specification for plugin support

### PR DESCRIPTION
**Pull request by smallsamantha**
This PR is a work in progress, open questions need to be addressed before full review.

- [ ] How should a Plugin indicate it supports Generated SDKs? 
- [ ] How should a Plugin indicate it is verified/managed? 
- [ ] How should a Plugin [and Module] represent its license?
- [ ] Should a filter on deprecated Plugins be provided on ListPlugins?
- [ ] Should a filter on Generated SDK support be provided on ListPlugins?
- [ ] Should an Update endpoint be provided?
- [ ] How should revision be validated?
- [ ] How should semver version be validated?
- [ ] How should digest be validated?
- [ ] Should it be specified that Modules/Plugins share a namespace (so names across both types must be unique per owner)?
- [ ] How should individual, language-specific Plugin configuration be represented?

### CODE REVIEW

File: buf/registry/plugin/v1/plugin_service.proto [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+package buf.registry.plugin.v1;
+
+import "buf/registry/owner/v1/owner.proto";
+import "buf/registry/plugin/v1/plugin.proto";
+import "buf/validate/validate.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/plugin/v1";
+
+// Operate on Plugins.
+service PluginService {
```
Comment (mfridman) [inline buf/registry/plugin/v1/plugin_service.proto:26]: Not sure if we're planning to bring in the [ResolveService](https://buf.build/bufbuild/buf/docs/main:buf.alpha.registry.v1alpha1#buf.alpha.registry.v1alpha1.ResolveService), but it'd be nice if we stubbed out a single RPC to resolve Generated SDK version. Something like

```
rpc ResolveGeneratedSDKVersion
```

Basically folding all the `Get{Go,Swift,Maven,NPM,...}Version` into a single RPC. You could argue that this fits nicely with the "PluginService".

Comment (saquibmian) [reply buf/registry/plugin/v1/plugin_service.proto:26]: Ideally there's also an RPC to get a registry-specific package ID. Could be the same RPC.

File: buf/registry/plugin/v1/plugin.proto [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The time the Plugin was created on the BSR.
+  google.protobuf.Timestamp create_time = 2 [(buf.validate.field).required = true];
+  // The last time the Plugin was updated on the BSR.
+  google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
+  // The name of the Plugin.
+  //
+  // Unique within a given User or Organization.
```
Comment (saquibmian) [inline buf/registry/plugin/v1/plugin.proto:40]: Does this imply that each version/revision of the plugin has a different name?

File: buf/registry/plugin/v1/plugin.proto [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    option (buf.validate.oneof).required = true;
+    // The id of the Plugin.
+    string id = 1 [(buf.validate.field).string.tuuid = true];
+    // The fully-qualified name of the Plugin.
+    Name name = 2;
+  }
+  // Semver-formatted release version of the Plugin.
+  string version = 3;
+  // The revision of the Plugin. This represents the BSR image version of the Plugin.
+  uint32 revision = 4;
```
Comment (saquibmian) [inline buf/registry/plugin/v1/plugin.proto:129]: Should these be under `Name`? Why would I want to specify these if I have the plugin ID?

**Review by saquibmian — COMMENTED** [review]
Drive by

**Pull request closed without merging**